From 521ede3fb735f20e586aabc827a5a3e3e1fa4b2a Mon Sep 17 00:00:00 2001
From: Julien Moutte <julien@fluendo.com>
Date: Fri, 27 Mar 2015 18:25:16 +0100
Subject: [PATCH] Remove prefix from events.

Youtube tests for EME are not handling prefixing properly, unprefix.
---
 Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp | 6 +++---
 Source/WebCore/Modules/encryptedmedia/MediaKeySession.h   | 6 +++---
 Source/WebCore/Modules/encryptedmedia/MediaKeySession.idl | 6 +++---
 Source/WebCore/dom/EventNames.h                           | 4 ++++
 4 files changed, 13 insertions(+), 9 deletions(-)

diff --git a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp
index 6e2b331..df53877 100644
--- a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp
+++ b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp
@@ -175,7 +175,7 @@ void MediaKeySession::addKeyTimerFired(Timer<MediaKeySession>*)
 
         // 2.7. If did store key is true, queue a task to fire a simple event named keyadded at the MediaKeySession object.
         if (didStoreKey) {
-            RefPtr<Event> keyaddedEvent = Event::create(eventNames().webkitkeyaddedEvent, false, false);
+            RefPtr<Event> keyaddedEvent = Event::create(eventNames().keyaddedEvent, false, false);
             keyaddedEvent->setTarget(this);
             m_asyncEventQueue->enqueueEvent(keyaddedEvent.release());
         }
@@ -211,7 +211,7 @@ void MediaKeySession::sendMessage(Uint8Array* message, String destinationURL)
     init.cancelable = false;
     init.message = message;
     init.destinationURL = destinationURL;
-    RefPtr<MediaKeyMessageEvent> event = MediaKeyMessageEvent::create(eventNames().webkitkeymessageEvent, init);
+    RefPtr<MediaKeyMessageEvent> event = MediaKeyMessageEvent::create(eventNames().keymessageEvent, init);
     event->setTarget(this);
     m_asyncEventQueue->enqueueEvent(event.release());
 }
@@ -221,7 +221,7 @@ void MediaKeySession::sendError(CDMSessionClient::MediaKeyErrorCode errorCode, u
     RefPtr<MediaKeyError> error = MediaKeyError::create(errorCode, systemCode).get();
     setError(error.get());
 
-    RefPtr<Event> keyerrorEvent = Event::create(eventNames().webkitkeyerrorEvent, false, false);
+    RefPtr<Event> keyerrorEvent = Event::create(eventNames().keyerrorEvent, false, false);
     keyerrorEvent->setTarget(this);
     m_asyncEventQueue->enqueueEvent(keyerrorEvent.release());
 }
diff --git a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.h b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.h
index 3f58510..e868fa6 100644
--- a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.h
+++ b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.h
@@ -73,9 +73,9 @@ class MediaKeySession : public RefCounted<MediaKeySession>, public EventTarget,
     // ActiveDOMObject
     virtual bool hasPendingActivity() const OVERRIDE;
 
-    DEFINE_ATTRIBUTE_EVENT_LISTENER(webkitkeyadded);
-    DEFINE_ATTRIBUTE_EVENT_LISTENER(webkitkeyerror);
-    DEFINE_ATTRIBUTE_EVENT_LISTENER(webkitkeymessage);
+    DEFINE_ATTRIBUTE_EVENT_LISTENER(keyadded);
+    DEFINE_ATTRIBUTE_EVENT_LISTENER(keyerror);
+    DEFINE_ATTRIBUTE_EVENT_LISTENER(keymessage);
 
     virtual const AtomicString& interfaceName() const OVERRIDE;
     virtual ScriptExecutionContext* scriptExecutionContext() const OVERRIDE { return ActiveDOMObject::scriptExecutionContext(); }
diff --git a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.idl b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.idl
index 7579565..0903d59 100644
--- a/Source/WebCore/Modules/encryptedmedia/MediaKeySession.idl
+++ b/Source/WebCore/Modules/encryptedmedia/MediaKeySession.idl
@@ -40,9 +40,9 @@
     void close();
     
     // EventListeners
-    attribute EventListener onwebkitkeyadded;
-    attribute EventListener onwebkitkeyerror;
-    attribute EventListener onwebkitkeymessage;
+    attribute EventListener onkeyadded;
+    attribute EventListener onkeyerror;
+    attribute EventListener onkeymessage;
 
     // EventTarget interface
     void addEventListener(DOMString type, 
diff --git a/Source/WebCore/dom/EventNames.h b/Source/WebCore/dom/EventNames.h
index b79b996..a55fa75 100644
--- a/Source/WebCore/dom/EventNames.h
+++ b/Source/WebCore/dom/EventNames.h
@@ -185,6 +185,10 @@ namespace WebCore {
     macro(webkitsourceended) \
     macro(webkitsourceclose) \
     \
+    macro(keyadded) \
+    macro(keyerror) \
+    macro(keymessage) \
+    macro(needkey) \
     macro(webkitkeyadded) \
     macro(webkitkeyerror) \
     macro(webkitkeymessage) \
